modules/javafx.controls/src/main/java/javafx/scene/control/DatePicker.java

```diff
@@ -147,11 +148,23 @@ public DatePicker(LocalDate localDate) {
 
         focusedProperty().addListener(o -> {
             if (!isFocused()) {
-                commitValue();
+                commitValueOnFocusLost();
             }
         });
     }
 
+    /**
+     * This is to address https://bugs.openjdk.org/browse/JDK-8303478
+     */
+    protected void commitValueOnFocusLost(){
```

Comment:
The `protected` modifier makes this method public API, which would most likely require a CSR to be approved. Since there doesn't seem to be a reason why this method should be public, I suggest making it private instead.

Reply:
ok, I have changed it to private